python/triqs/operators/util/U_matrix.py

```diff
@@ -131,43 +135,88 @@ def reduce_4index_to_2index(U_4index):
 
     return U, Uprime
 
-# Construct the 2-index matrices for the density-density form
-def U_matrix_kanamori(n_orb, U_int, J_hund):
+def U_matrix_kanamori(n_orb, U_int, J_hund, Up_int=None, full_Uijkl=False, Jc_hund=None):
     r"""
-    Calculate the Kanamori U and Uprime matrices.
+    Calculate the Kanamori two-index interaction matrix for parallel spins:
+
+    .. math:: U_{m m'}^{\sigma \sigma} \equiv U_{m m' m m'} - J_{m m'}
+
+    with:
+
+    .. math:: J_{m m'} \equiv U_{m m' m' m} ,
+
+    and the two-index interaction matrix for anti-parallel spins:
+
+    .. math:: U_{m m'}^{\sigma \bar{\sigma}} \equiv U_{m m' m m'}
+
+    If full_Uijkl=True is specified instead the full four index 
+    Uijkl tensor is returned. 
 
     Parameters
     ----------
     n_orb : integer
             Number of orbitals in basis.
-    U_int : scalar
+    U_int : float
             Value of the screened Hubbard interaction.
-    J_hund : scalar
+    J_hund : float
              Value of the Hund's coupling.
+    Up_int : float, optional
+            Value of the screened U prime parameter
+            defaults to U_int-2*J_hund if not given.
+            (fully rotationally invariant form)
+    full_Uijkl : bool, optional
+            retunr instead the full four-index Uijkl tensor
+            default is False
+    Jc_hund : foat, optional
+            only used if full_Uijkl=True, defaults to J_hund
 
     Returns
     -------
     U : float numpy array
-        The two-index interaction matrix for parallel spins.
+        The two-index interaction matrix for parallel spins or
+        the four-index Uijkl tensor if full_Uijkl=True
     Uprime : float numpy array
-             The two-index interaction matrix for anti-parallel spins.
+        The two-index interaction matrix for anti-parallel spins.
 
     """
 
-    U  = np.zeros((n_orb,n_orb),dtype=float)      # matrix for same spin
-    Uprime = np.zeros((n_orb,n_orb),dtype=float)  # matrix for opposite spin
+    # Jc_hund can only be used if the full tensor is returned
+    if Jc_hund is not None and not full_Uijkl:
+        raise ValueError('Jc_hund can only be specified if the full four index tensor is returned')
 
-    m_range = list(range(n_orb))
-    for m,mp in product(m_range,m_range):
-        if m == mp:
-            Uprime[m,mp] = U_int
-        else:
-            U[m,mp]  = U_int - 3.0*J_hund
-            Uprime[m,mp] = U_int - 2.0*J_hund
+    if not Up_int:
```

Comment:
`not Up_int` evaluates to `True` for `Up_int==0.0`. Is this the desired behavior?

Reply:
addressed in e6725183